src/lib/remark-plugins/rewrite-tutorial-links/index.ts

```diff
@@ -30,29 +30,60 @@ import {
 
 let TUTORIAL_MAP
 
-export const rewriteTutorialLinksPlugin: Plugin = () => {
+const DEFAULT_CONTENT_TYPE = 'tutorials'
+
+interface RewriteTutorialLinksPluginOptions {
+	contentType?: 'docs' | 'tutorials'
+	tutorialMap?: Record<string, string>
+}
+
+export const rewriteTutorialLinksPlugin: Plugin = (
+	options: RewriteTutorialLinksPluginOptions = {}
+) => {
+	const { contentType = DEFAULT_CONTENT_TYPE, tutorialMap } = options
 	return async function transformer(tree) {
-		TUTORIAL_MAP = await getTutorialMap()
+		// Load the tutorial map if it's not provided
+		TUTORIAL_MAP = tutorialMap ?? (await getTutorialMap())
 
-		visit(tree, 'link', handleRewriteTutorialsLink)
-		visit(tree, 'definition', handleRewriteTutorialsLink)
+		// Visit link and defintion node types
+		visit(tree, ['link', 'definition'], (node: Link | Definition) => {
+			handleRewriteTutorialsLink(node, contentType)
+		})
 	}
 }
 
-function handleRewriteTutorialsLink(node: Link | Definition) {
-	node.url = rewriteTutorialsLink(node.url, TUTORIAL_MAP)
+function handleRewriteTutorialsLink(
+	node: Link | Definition,
+	contentType: RewriteTutorialLinksPluginOptions['contentType'] = DEFAULT_CONTENT_TYPE
+) {
+	node.url = rewriteTutorialsLink(node.url, TUTORIAL_MAP, contentType)
+}
+
+const isRelativeUrl = (url) => {
+	try {
+		new URL(url)
+		return false
+	} catch (e) {
+		return true
+	}
 }
 
 export function rewriteTutorialsLink(
 	url: string,
-	tutorialMap: Record<string, string>
+	tutorialMap: Record<string, string>,
+	contentType: RewriteTutorialLinksPluginOptions['contentType'] = DEFAULT_CONTENT_TYPE
 ): string {
 	let newUrl
 
 	try {
 		const urlObject = new URL(url, 'https://learn.hashicorp.com')
 
-		const isExternalLearnLink = getIsExternalLearnLink(url)
+		let isExternalLearnLink
+		if (contentType === 'docs') {
+			isExternalLearnLink = !isRelativeUrl(url)
```

Comment:
We should check if it _is_ an external learn link right? and not only that its not a relative docs link? for example, if theres a learn.hashicorp.com link in docs content, we still want to rewrite that. 

Also, if "https://kubernetes.io/docs/home/" were passed here, wouldn't `isExternalLearnLink` be true? Or are we only matching on 'ok' domains elsewhere?

Reply:
I think this'll need a slight adjustment, brb! Will add some more test cases to double-check

Reply:
@kendallstrautman updated!

Reply:
Looks good!